lib/libzfs/libzfs_dataset.c

```diff
@@ -4354,6 +4355,18 @@ zfs_rename(zfs_handle_t *zhp, const char *target, boolean_t recursive,
 }
 
 nvlist_t *
+zfs_get_all_props(zfs_handle_t *zhp)
+{
+	return (zhp->zfs_props);
+}
+
+nvlist_t *
+zfs_get_recvd_props(zfs_handle_t *zhp)
+{
+	return (zhp->zfs_recvd_props);
+}
```

Comment:
I finally went and looked at the OpenZFS source for [this function](https://github.com/openzfs/openzfs/blob/master/usr/src/lib/libzfs/common/libzfs_dataset.c#L4228) 
which still exists (although it's unused).  We really should use that version which properly fetches the properties if needed.  In the case of `zfs_get_all_props()` it should be valid like `zfs_get_user_props()`.

Reply:
So replace ```zfs_get_recvd_props()``` with the following implementation?

https://github.com/openzfs/openzfs/blob/5fecb44ac1357a01041c95e3de674cd8b882852f/usr/src/lib/libzfs/common/libzfs_dataset.c#L4228-L4235

Reply:
Yes.

Reply:
Went ahead and rebased off master too while I was at it.

Why isn't that necessary for the other two functions?

Reply:
This should be safe for `zhp->zfs_props` and `zhp->zfs_user_props` because they're both populated when the dataset is opened as part of `zfs_open()`.

```
zfs_open()
  make_dataset_handle()
    make_dataset_handle_common()
      put_stats_zhdl()  <- set in here.
```